### ISSUE TITLE
.NET 8 Preview 7 Blazor WebAssembly PWA, not working on iPhone & iPad

### ISSUE THREAD

**Issue by atuggle** [dotnet/runtime#90710]
## Description

After upgrading a .NET 7 Blazor WebAssembly PWA application to .NET 8 Preview 7 I noticed the web app stopped loading on iPhones and iPads.  The console log shows the below error:
```
[Error] Error: AggregateException_ctor_DefaultMessage (Arg_IndexOutOfRangeException) — dotnet.runtime.8.0.0-preview.7.23375.6.hh1fgtyrm9.js:3:29648
	(anonymous function) (blazor.webassembly.js:1:40595)
```

While attempting to create a sample, I discovered the two lines of code that reproduce the bug.  The sample project with README.md instructions can be found here: https://github.com/atuggle/BlazorWebAssemblyPWA-BreaksOnIOS 

- Build and deploy the above repository to duplicate it on iPhone
- The repo has 2 commits, the second commits are the breaking changes that must both be added

## Configuration

There are two steps to reproduce this and it's pretty easy.  Step 1 create a sample Blazor WebAssemby PWA project, and Step 2 add two lines of code deploy, and verify on iPhone it does not work.

### Step 1: Create the project

1.  Open Visual Studio 2022 Preview Edition
2.  Choose "New Project" to create a new project
3.  Find and choose "Blazor WebAssembly App"
4.  Name the App and choose "Next"
5.  Ensure ".NET 8.0 (Preview)" is selected as the framework and check "Progressive Web Application"

### Step 2: Add code that breaks it on iPhone & iPad

1. Turn on AOT by adding 
    ```
    <RunAOTCompilation>true</RunAOTCompilation> to the project file
    ```
2. Add an <InputText .../> element in the index.razor page
    ```
    <InputText id="emailText" class="form-control" />
    ```
3. Deploy the above code and see it fails on IOS with the error

### Other information

I am able to work around this right now by turning AOT off, and most people will not notice this because AOT is off by default for most new project creations.

**Comment by pavelsavara** [comment]
I don't have any apple device, so I just tested the provided repository with chrome on windows.
I get 

![image](https://github.com/dotnet/runtime/assets/271576/92f2ba78-6aae-49a0-98bc-1c5795ab51d8)


Also, this is compiled with SIMD instructions and perhaps your devices doesn't have recent version of Safari (16.4 and above) ?

**Comment by pavelsavara** [comment]
> I added a new [commit](https://github.com/atuggle/BlazorWebAssemblyPWA-BreaksOnIOS/commit/c694d8f753fffb2c1dd301c4f4683a0b1ea9e5fe) that fixes the InputText for chrome.

The new version just works on my chrome, so this need somebody with ios device to have look.
Does it also fail on OsX ?

>     * Note: This does not impact the issue where AOT compilation breaks the app on Iphones and Ipads

Now I'm thinking that if this was SIMD or EH issue it would fail much earlier.

Could you please try to give us more detailed stack trace ?

And also please report the version of Safari on the device.

Thanks!

**Comment by atuggle** [comment]
It does work on OsX!  Just not on iPhones and iPads!

The device I'm using is on an iphone running IOS 16.6 which is also the version of Safari for IOS.

**Comment by atuggle** [comment]
Safari will not allow me to copy the stack trace so the following screen captures show the stack trace:
![image](https://github.com/dotnet/runtime/assets/2790037/3f958fa3-d05a-4a63-837a-7f8b08823fa8)
![image](https://github.com/dotnet/runtime/assets/2790037/e4444134-6451-45a3-841a-a178e19cdd35)
![image](https://github.com/dotnet/runtime/assets/2790037/02200d3d-b696-4e7d-96ca-f6bad5ec105f)
![image](https://github.com/dotnet/runtime/assets/2790037/dfe8aec3-503e-427b-8aeb-22001a94cd50)

**Comment by kg** [comment]
Does this reproduce with the Jiterpreter disabled?
```xml
<PropertyGroup>
   <BlazorWebAssemblyJiterpreter>false</BlazorWebAssemblyJiterpreter>
</PropertyGroup>
```

**Comment by pavelsavara** [comment]
Could you please enable symbols
```xml
<PropertyGroup>
  <WasmNativeDebugSymbols>true</WasmNativeDebugSymbols>
  <WasmNativeStrip>false</WasmNativeStrip>
</PropertyGroup>
```

And see if stack trace has better function names, if so please copy them again.

**Comment by atuggle** [comment]
![image](https://github.com/dotnet/runtime/assets/2790037/0144f09a-b5a2-4b16-98ea-4faf6a11463f)
![image](https://github.com/dotnet/runtime/assets/2790037/e0c30253-05a4-4c65-b445-f66257872c10)
![image](https://github.com/dotnet/runtime/assets/2790037/983c56ff-89ed-45ab-bbc6-ed9eb9b34ad4)
![image](https://github.com/dotnet/runtime/assets/2790037/15832623-55ec-4f28-a0d6-4854ffc1bc1a)

Hope this helps

**Comment by atuggle** [comment]
Additional context:  It doesn't work in Chrome on my iPhone either.  So both Safari & Chrome do not work when I have AOT = true on iPhone & iPad.

**Comment by atuggle** [comment]
This is the resulting page on my iPhone:
![image](https://github.com/dotnet/runtime/assets/2790037/edd5a4c0-e721-4d06-bb5e-2d5c3cb4f5a4)

**Comment by kg** [comment]
> Additional context: It doesn't work in Chrome on my iPhone either. So both Safari & Chrome do not work when I have AOT = true on iPhone & iPad.

FYI, Chrome and Firefox on iPhone/iPad are just Safari in a wrapper. So you don't need to test with them.

**Comment by pavelsavara** [comment]
I think the top one screen is just JS interop rejecting a promise. Something happened deeper than that.

**Comment by kobealbright** [comment]
Similar issues here on .Net 8 RC1.

**Comment by SpikeViper** [comment]
Same issue in .Net 8 RC1. Need fix.

**Comment by e012345678** [comment]
Is this fixed in .NET 8 ?

**Comment by GlitchedPolygons** [comment]
> Is this fixed in .NET 8 ?

Nope, still sucks :')

**Comment by e012345678** [comment]
Any news on this? Is this working in .NET 8.0.1 then?

**Comment by GlitchedPolygons** [comment]
> Any news on this? Is this working in .NET 8.0.1 then?

New year, old problems. 

Doesn't work for me yet... I had to add

```
<RunAOTCompilation>false</RunAOTCompilation>
<WasmEnableSIMD>false</WasmEnableSIMD>
```

to the `.csproj` file's `PropertyGroup` section to temporarily circumnavigate this situation...

**Comment by e012345678** [comment]
Thanks for the reply @GlitchedPolygons, we would like to use AOT however, guess that's not possible then?

**Comment by GlitchedPolygons** [comment]
> Thanks for the reply @GlitchedPolygons, we would like to use AOT however, guess that's not possible then?

I'd love to use AOT too (and the SIMD optimizations would be very welcome too since I'm working with relatively math-intensive stuff...) but unfortunately at the moment it doesn't look like we're getting a fix anytime soon :(

**Comment by e012345678** [comment]
Any comment on this from the Microsoft team? @pavelsavara

**Comment by kg** [comment]
If `<WasmEnableSIMD>false</WasmEnableSIMD>` fixes it in interpreter mode, that suggests the device you're running on lacks SIMD support. It sounds like it doesn't work *at all* in AOT mode though even if SIMD is disabled, correct? That sounds like a separate issue with the AOT compiler.

I personally don't own any Apple hardware other than a macbook so I can't test/verify this problem myself.

If my understanding is correct, what remains of this issue is the AOT problem. It's expected that if SIMD is enabled you won't be able to run on old devices that lack SIMD support. (Whether SIMD should be disabled by default is a separate question which would need a separate tracking issue.)

**Comment by e012345678** [comment]
We are using these two lines in our csproj in .NET 7 and it works perfectly, what is different in .NET 8?

```
    <WasmEnableSIMD Condition="'$(Configuration)'=='Release'">true</WasmEnableSIMD>
    <RunAOTCompilation Condition="'$(Configuration)'=='Release'">true</RunAOTCompilation>
```

This seems to work even if the device doesn't have SIMD support.

**Comment by lewing** [comment]
please add iOS version information

**Comment by GlitchedPolygons** [comment]
Bump.

**Comment by maraf** [comment]
@e012345678 @GlitchedPolygons What version of iOS/Safari are you seeing this message one? 
And are you getting the `Error: AggregateException_ctor_DefaultMessage`, right?

**Comment by GlitchedPolygons** [comment]
> @e012345678 @GlitchedPolygons What version of iOS/Safari are you seeing this message one? And are you getting the `Error: AggregateException_ctor_DefaultMessage`, right?

Ahhw, I don't remember >.<

I do think so tho, yeah... It does ring a bell.

(I had to rely on the above workaround of disabling SIMD and AOT comp. to make it work)

**Comment by maraf** [comment]
My current testing
- Running the sample repo on Mac M1 in Safari or Edge works (both have SIMD)
- Running on iOS simulator 15.5 in Safari doesn't work (doesn't have SIMD support)
- Disabling SIMD (`WasmEnableSIMD=false`) makes it work on iOS simulator 15.5 in Safari
- In any case I'm not able to get `Error: AggregateException_ctor_DefaultMessage`

**Comment by maraf** [comment]
The sample repo uses preview of .NET 8. 
Can you please validate that AOT compilation works in latest .NET 8 (8.0.3) in your environment?

**Comment by atuggle** [comment]
I am pleased to report this issue is resolved (specifically the `Error: AggregateException_ctor_DefaultMessage` issue). 
I have verified this specific issue has been resolved using .NET 8 (8.0.100) with the same devices I originally reported this issue with. My personal iPad and iPhone.  Also, as before my application continues to works on multiple OSX devices including several older model Macbook pro's and a newer M3 based Macbook pro.

I am going to leave this Issue Open since there appears to be other apps with similar issues, but you are free to close this since I can verify my specific issue works (Event with RunAOTCompilation = true)
